code/modules/clothing/chameleon/chameleon_drone.dm

```diff
@@ -0,0 +1,59 @@
+/datum/action/item_action/chameleon/drone/randomise
+	name = "Randomise Headgear"
+	button_icon = 'icons/mob/actions/actions_items.dmi'
+	button_icon_state = "random"
+
+/datum/action/item_action/chameleon/drone/randomise/Trigger(trigger_flags)
+	if(!IsAvailable(feedback = TRUE))
+		return FALSE
+
+	for(var/datum/action/item_action/chameleon/change/to_randomize in owner.actions)
+		to_randomize.random_look()
+	return TRUE
+
+/datum/action/item_action/chameleon/drone/togglehatmask
+	name = "Toggle Headgear Mode"
+	button_icon = 'icons/mob/actions/actions_silicon.dmi'
+
+/datum/action/item_action/chameleon/drone/togglehatmask/New()
+	. = ..()
+
+	if (istype(target, /obj/item/clothing/head/chameleon/drone))
+		button_icon_state = "drone_camogear_helm"
+	if (istype(target, /obj/item/clothing/mask/chameleon/drone))
+		button_icon_state = "drone_camogear_mask"
+
+/datum/action/item_action/chameleon/drone/togglehatmask/Trigger(trigger_flags)
+	if(!IsAvailable(feedback = TRUE))
+		return
+
+	// No point making the code more complicated if no non-drone
+	// is ever going to use one of these
+
+	var/mob/living/simple_animal/drone/D
+
+	if(isdrone(owner))
+		D = owner
+	else
+		return
```

Comment:
early return? i'm nitting this even though it's clearly reorganized code because this is being billed as a refactor

Reply:
the code is also really strange in that we typecast _first_, then typecheck. zamn